src/integTest/kotlin/io/snyk/plugin/ui/toolwindow/settings/ScanTypesPanelTest.kt

```diff
@@ -0,0 +1,147 @@
+    }
+
+    override fun tearDown() {
+        unmockkAll()
+        resetSettings(project)
+        disposable.dispose()
+        // restore modified Registry value
+        isContainerEnabledRegistryValue.setValue(isContainerEnabledDefaultValue)
+        super.tearDown()
+    }
+
+    private lateinit var disposable: Disposable
+
+    private val isContainerEnabledRegistryValue = Registry.get("snyk.preview.container.enabled")
+    private val isContainerEnabledDefaultValue: Boolean by lazy { isContainerEnabledRegistryValue.asBoolean() }
+
+    private fun setUpContainerTest() {
+        isContainerEnabledRegistryValue.setValue(true)
+    }
+
+    @Test
+    fun `container scan enablement get from settings`() {
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = true
+
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        assertTrue("", containerCheckBox.isSelected)
+    }
+
+    @Test
+    fun `container scan enablement set to settings`() {
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = true
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        containerCheckBox.doClick()
+        scanTypesPanel.panel.apply()
+
+        assertFalse("", settings.containerScanEnabled)
+    }
+
+    @Test
+    fun `container scan disablement get from settings`() {
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = false
+
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        assertFalse("", containerCheckBox.isSelected)
+    }
+
+    @Test
+    fun `container scan disablement set to settings`() {
+        setUpContainerTest()
+        val settings = pluginSettings()
+        settings.containerScanEnabled = false
+        val scanTypesPanel = ScanTypesPanel(project, disposable)
+        val containerCheckBox =
+            getComponentByName(scanTypesPanel.panel, JBCheckBox::class, "containerEnablementCheckBox")
+                ?: throw IllegalStateException("containerEnablementCheckBox not found")
+
+        containerCheckBox.doClick()
+        scanTypesPanel.panel.apply()
+
+        assertTrue("", settings.containerScanEnabled)
```

Comment:
see above. Just for the sake of mentioning it - we do have some code duplication here, so one option would be to extract the common parts, a second option would be to use a parametrized test (see https://www.tutorialspoint.com/junit/junit_parameterized_test.htm)